frontend/src/components/game/PlayerGameView.tsx

```diff
@@ -307,22 +366,48 @@ function PlayerGameView(props: PlayerGameViewProps) {
 
   return (
     <>
-      {!spectatorUnsubscribePlayer ? (
+      {!spectatorUnsubscribePlayer && !spectateGame ? (
         <LeaderboardContent>
           {displayPlayerLeaderboard()}
         </LeaderboardContent>
       ) : (
-        <>
-          <p>
-            Spectate
-            {' '}
-            {spectateGame?.player.nickname}
-          </p>
-
-          <PrimaryButton onClick={spectatorUnsubscribePlayer}>
-            Go Back
-          </PrimaryButton>
-        </>
+        <GameHeaderContainer>
+          <GameHeaderContainerChild>
+            <SpectatorBackIcon
+              onClick={spectatorUnsubscribePlayer || (() => {})}
+            >
+              arrow_back
+            </SpectatorBackIcon>
```

Comment:
Done! There are some components named Icon that are not in this file, because they are slightly different than the rest and not reused; however, I put a number of components (including this one) in the new `Icon.tsx` file.